builtin/providers/softlayer/resource_softlayer_virtual_guest.go

```diff
@@ -194,29 +194,33 @@ func getBlockDevices(d *schema.ResourceData) []datatypes.BlockDevice {
 	}
 }
 
-func getPostProvisioningHookFromResourceData(d *schema.ResourceData, meta interface{}) (datatypes.SoftLayer_Provisioning_Hook, error) {
+func getPostProvisioningHooksByName(provisioningHookName string, meta interface{}) ([]datatypes.SoftLayer_Provisioning_Hook, error) {
         client := meta.(*Client).accountService
 
         mask := []string{
                 "id",
+                "name",
+                "uri",
         }
-
-        provisioningHookName := d.Get("provisioning_hook_name").(string)
-
-        filter := fmt.Sprintf(`{"name":"%s"}`, provisioningHookName)
-
-        hooks, err := client.GetPostProvisioningHooks(mask, filter)
+        
+        filter := fmt.Sprintf(`{"postProvisioningHooks":{"name":{"operation":"%s"}}}`, provisioningHookName)
+        
+        provisioningHooks, err := client.GetPostProvisioningHooks(mask, filter)
 
         if err != nil {
-                return datatypes.SoftLayer_Provisioning_Hook{}, fmt.Errorf("Error looking up Provisioning Hook: %d %s", len(hooks), err)
+                return nil, fmt.Errorf("Error looking up Provisioning Hook: %s", err)
         }
 
-        if len(hooks) < 1 {
-                return datatypes.SoftLayer_Provisioning_Hook{}, fmt.Errorf(
+        if len(provisioningHooks) < 1 {
+                return nil, fmt.Errorf(
                         "Unable to locate a provisioning hook matching the provided provisioning hook name: %s", provisioningHookName)
         }
+        
+        for _, hook := range provisioningHooks {
+                log.Printf("[INFO] The retrieved post provisioning hook name and uri : %s, %s", hook.Name, hook.Uri)
+        }
 
-        return hooks[0], nil
+        return provisioningHooks, nil
```

Comment:
And the name of the function should be changed to the singular getPostProvisioningHookByName